jena-integration-tests/src/test/java/org/apache/jena/test/rdfconnection/TestRDFConnectionRemote.java

```diff
@@ -228,11 +230,27 @@ public void non_standard_syntax_update_remote_3c() {
     /** Standard update syntax with substitution should work. */
     @Test
     public void standard_syntax_update_remote_1a() {
+        RDFNode FALSE = ResourceFactory.createTypedLiteral(false);
         try ( RDFConnection conn = RDFConnectionRemote.service(server.datasetURL("/ds")).parseCheckSPARQL(false).build() ) {
             conn.newUpdate()
-                .update("INSERT DATA { <a> <b> <c> }")
-                .update("INSERT DATA { <x> <y> <z> }")
-                .substitution("foo", RDF.type).build();
+                .update("INSERT { <a> <b> <c> } WHERE { FILTER(?foo) }")
+                .update("INSERT { <x> <y> <z> } WHERE { FILTER(?foo) }")
+                .substitution("foo", FALSE)
+                .build()
+                .execute();
+        }
+    }
+
+    /** Standard update syntax with substitution should work when comments are involved. */
+    @Test
+    public void standard_syntax_update_remote_2b() {
```

Comment:
Strictly speaking, this additional test case could go into a separate commit. It tests for an oversight I introduced with #2580 where update strings were joined on the semicolon without a preceding newline. In such cases, joining update strings that ended on a comment would produce syntax errors.